chain/chain/src/resharding/resharding_actor.rs

```diff
@@ -1,23 +1,88 @@
-use super::types::FlatStorageSplitShardRequest;
-use near_async::messaging::{self, Handler};
+use super::types::{
+    FlatStorageShardCatchupRequest, FlatStorageSplitShardRequest, MemtrieReloadRequest,
+};
+use crate::flat_storage_resharder::{FlatStorageResharder, FlatStorageReshardingTaskStatus};
+use crate::ChainStore;
+use near_async::futures::{DelayedActionRunner, DelayedActionRunnerExt};
+use near_async::messaging::{self, Handler, HandlerWithContext};
+use near_primitives::hash::CryptoHash;
+use near_primitives::types::BlockHeight;
+use near_store::{ShardUId, Store};
+use time::Duration;
 
 /// Dedicated actor for resharding V3.
-pub struct ReshardingActor {}
+pub struct ReshardingActor {
+    chain_store: ChainStore,
+}
 
 impl messaging::Actor for ReshardingActor {}
 
 impl Handler<FlatStorageSplitShardRequest> for ReshardingActor {
     fn handle(&mut self, msg: FlatStorageSplitShardRequest) {
-        self.handle_flat_storage_split_shard_request(msg);
+        match msg.resharder.split_shard_task() {
+            FlatStorageReshardingTaskStatus::Successful { .. } => {
+                // All good.
+            }
+            FlatStorageReshardingTaskStatus::Failed => {
+                panic!("impossible to recover from a flat storage split shard failure!")
+            }
```

Comment:
nit: It may be beneficial to provide some extra info in the Failed and in the panic message. I guess you log it directly before returning failed but still wouldn't hurt.